### DISCUSSION TITLE
GPL applicability to extensions

### DISCUSSION THREAD

**Discussion by DigNZ** [directus/directus#14422] [Archive]
Hi,

Just checking if custom extensions build using the extensions SDK are caught up with the GPL or another licence? I've got some integrations with third party APIs that are implemented as extensions (hooks) on a schedule and populate the data via some manipulations and transformation and then store it in the directus database for API and admin access. 

If we distribute this solution to customers we have no problem crediting and supporting directus if we make modifications to the core code but the data manipulations are propritery so need we need to protect the extensions. 

If they ARE covered by the GPL because they use the directus/extensions-sdk I'll need to rewrite them as node modules next to directus and put the data straight into the database.

**Comment by rijkvanzanten** [comment]
_Disclaimer: I'm not a lawyer_

As far as I understand (and again, this is all very confusing legal stuff), _usage_ of a distributed build of the package (like `directus` or the extensions-sdk) isn't the same as making a copy of the source code and editing that. The latter would require to re-use the same license and attribute the original source, but as far as I'm aware / concerned, a new project that's _using_ the exposed APIs isn't touched by that. Happy to keep discussing!
